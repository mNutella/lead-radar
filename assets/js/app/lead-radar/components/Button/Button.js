import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

import './style.scss';


const renderLinkBtn = (children, link, classes) => (
  <a
    className={`btn btn-block p-3 ${classes.join(' ')}`}
    href={link}
    role="button"
  >
    {children}
  </a>
);

const renderRouteBtn = (children, link, classes) => (
  <Link
    to={link}
    role="button"
    className={`btn btn-block p-3 ${classes.join(' ')}`}
  >
    {children}
  </Link>
);

const renderDefBtn = (children, type, classes, onClick) => (
  // eslint-disable-next-line react/button-has-type
  <button
    type={type}
    className={`btn btn-block p-3 ${classes.join(' ')}`}
    onClick={onClick}
  >
    {children}
  </button>
);

const renderBtn = (children, route, def, link, type, classes, onClick) => {
  if (route) {
    return renderRouteBtn(children, link, classes);
  }
  if (def) {
    return renderDefBtn(children, type, classes, onClick);
  }
  if (!route && !def) {
    return renderLinkBtn(children, link, classes);
  }

  return <div>Unknown btn type</div>;
};

const Button = ({
  children, route, def, link, type, size, classes, onClick,
}) => (
  <div className="row">
    <div className={`col-sm-${size}`}>
      {renderBtn(children, route, def, link, type, classes, onClick)}
      {/* {route ? (
        <Link
          className={`btn btn-block p-3 ${classes.join(' ')}`}
          to={link}
          role="button"
          onClick={onClick}
        >
          {children}
        </Link>
      ) : (
        <a
          className={`btn btn-block p-3 ${classes.join(' ')}`}
          href={link}
          role="button"
          onClick={onClick}
        >
          {children}
        </a>
      )
      } */}
    </div>
  </div>
);

Button.defaultProps = {
  route: false,
  def: false,
  link: '#!',
  type: 'button',
  size: 12,
  children: undefined,
  classes: [],
  onClick: undefined,
};

Button.propTypes = {
  route: PropTypes.bool,
  def: PropTypes.bool,
  link: PropTypes.string,
  type: PropTypes.string,
  size: PropTypes.number,
  // eslint-disable-next-line react/forbid-prop-types
  children: PropTypes.any,
  // eslint-disable-next-line react/forbid-prop-types
  classes: PropTypes.any,
  // eslint-disable-next-line react/forbid-prop-types
  onClick: PropTypes.any,
};

export default Button;
