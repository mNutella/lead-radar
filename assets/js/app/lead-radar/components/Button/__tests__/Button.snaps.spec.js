import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from '../Button';

describe('<Button />', () => {
  test('render button', () => {
    const wrapper = shallow((
      <Button />
    ));

    expect(wrapper).toMatchSnapshot();
  });

  test('markup button type "Link"', () => {
    const link = '/home';
    const wrapper = mount((
      <Button link={link} />
    ));
    const button = wrapper.find(Button);

    expect(button.prop('link')).toBe(link);
    expect(button.children()).toMatchSnapshot();
  });

  test('markup button type "Def"', () => {
    const wrapper = mount((
      <Button def />
    ));
    const button = wrapper.find(Button);

    expect(button.prop('def')).toBe(true);
    expect(button.children()).toMatchSnapshot();
  });

  test('add one class', () => {
    const classes = ['radius'];
    const wrapper = mount((
      <Button classes={classes} />
    ));
    const button = wrapper.find(Button);

    expect(button.prop('classes')).toBe(classes);
    expect(button.children()).toMatchSnapshot();
  });

  test('add some classes', () => {
    const classes = ['radius', 'border-pill'];
    const wrapper = mount((
      <Button classes={classes} />
    ));
    const button = wrapper.find(Button);

    expect(button.prop('classes')).toBe(classes);
    expect(button.children()).toMatchSnapshot();
  });
});
