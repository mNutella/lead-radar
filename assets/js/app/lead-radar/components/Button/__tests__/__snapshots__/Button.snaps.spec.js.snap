// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> add one class 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <a
      className="btn btn-block p-3 radius"
      href="#!"
      role="button"
    />
  </div>
</div>
`;

exports[`<Button /> add some classes 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <a
      className="btn btn-block p-3 radius border-pill"
      href="#!"
      role="button"
    />
  </div>
</div>
`;

exports[`<Button /> markup button type "Def" 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <button
      className="btn btn-block p-3 "
      disabled={false}
      type="button"
    />
  </div>
</div>
`;

exports[`<Button /> markup button type "Link" 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <a
      className="btn btn-block p-3 "
      href="/home"
      role="button"
    />
  </div>
</div>
`;

exports[`<Button /> render button 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <a
      className="btn btn-block p-3 "
      href="#!"
      role="button"
    />
  </div>
</div>
`;
